### PR TITLE
Continuous integration on Travis

### PR DESCRIPTION
**Pull request by simogasp**
* added CI with caching of dependencies (Boost, Eigen, OpenCV)
* for now only CPU version supported (no CUDA)
* since we need CMake >= 3.4, for now it downloads the latest (3.6) and it caches it
* it compiles with g++-4.8 and clang-3.7

connected to #24

### CODE REVIEW

File: src/cctag/Detection.cpp [hunk 1]
```diff
@@ -876,7 +876,7 @@ void cctagDetection(CCTag::List& markers,
         }
 #endif // WITH_CUDA
 
-        std::vector<cctag::ImageCut> vSelectedCuts[ numTags ];
+        std::vector<std::vector<cctag::ImageCut> > vSelectedCuts( numTags );
```
Comment (simogasp) [inline src/cctag/Detection.cpp]: @griwodz pls check if it is a safe change, clang was somehow complaining about making a c-array of vectors. It compiles after this but I hope there are not hidden side effects..

**Comment by simogasp** [comment]
@lcalvet @griwodz @zvrba can someone check this (https://github.com/poparteu/CCTag/pull/27#discussion_r74903426) and merge?

**Comment by zvrba** [comment]
I've looked at the context and it looks fine. I'll merge it.

**Comment by simogasp** [comment]
thx!

**Pull request merged**